cpp/modmesh/universe/polygon.hpp

```diff
@@ -781,6 +787,577 @@ class TrianglePad
+        {
+            return 0;
+        }
+
+        value_type area = 0;
+        for (size_t i = 0; i < count; ++i)
+        {
+            point_type const p0 = m_nodes->get(offset + i);
+            point_type const p1 = m_nodes->get(offset + ((i + 1) % count));
+            area += p0.x() * p1.y() - p1.x() * p0.y();
+        }
+
+        return area / 2;
+    }
+
+    /**
+     * Check if polygon nodes are ordered counter-clockwise (right-hand rule).
+     */
+    bool is_counter_clockwise(size_t polygon_id) const
+    {
+        return compute_signed_area(polygon_id) > 0;
+    }
+
+    /**
+     * Calculate bounding box for a specific polygon.
+     */
+    BoundBox3d<T> calc_bound_box(size_t offset, size_t count) const
+    {
+        if (count == 0)
+        {
+            return BoundBox3d<T>(0, 0, 0, 0, 0, 0);
+        }
+
+        value_type min_x = std::numeric_limits<value_type>::max();
+        value_type min_y = std::numeric_limits<value_type>::max();
+        value_type min_z = std::numeric_limits<value_type>::max();
+        value_type max_x = std::numeric_limits<value_type>::lowest();
+        value_type max_y = std::numeric_limits<value_type>::lowest();
+        value_type max_z = std::numeric_limits<value_type>::lowest();
+
+        for (size_t i = 0; i < count; ++i)
+        {
+            point_type const node = m_nodes->get(offset + i);
+            min_x = std::min(min_x, node.x());
+            min_y = std::min(min_y, node.y());
+            min_z = std::min(min_z, node.z());
+            max_x = std::max(max_x, node.x());
+            max_y = std::max(max_y, node.y());
+            max_z = std::max(max_z, node.z());
+        }
+
+        return BoundBox3d<T>(min_x, min_y, min_z, max_x, max_y, max_z);
+    }
+
+    /**
+     * Search for segments within a bounding box across all polygons.
+     * Returns segments from all polygons that intersect the query box.
+     *
+     * @param box Query bounding box
+     * @param output Vector to store found segments
+     */
+    void search_segments(BoundBox3d<T> const & box, std::vector<segment_type> & output) const
+    {
+        m_rtree->search(box, output);
+    }
+
+    /**
+     * Rebuild the spatial index (RTree) for all polygons.
+     * Call this after modifying polygon data.
+     */
+    void rebuild_rtree()
+    {
+        m_rtree = std::make_unique<rtree_type>();
+        for (size_t i = 0; i < m_polygon_offsets.size(); ++i)
+        {
+            polygon_type polygon = get_polygon(i);
+            rebuild_polygon_rtree(polygon);
+        }
+    }
+
+private:
+
+    void rebuild_polygon_rtree(polygon_type const & polygon)
+    {
+        size_t const polygon_id = polygon.polygon_id();
+        size_t const count = polygon.count();
+
+        for (size_t i = 0; i < count; ++i)
+        {
+            segment_type const edge = get_edge(polygon_id, i);
+            m_rtree->insert(edge);
+        }
+    }
+
+    std::shared_ptr<point_pad_type> m_nodes;
```

Comment:
Rename `m_nodes` to `m_points` for consistency with the type names of `Point3d` and `PointPad`.

Reply:
ok